### PR TITLE
Fixes for Identity 3552, Identity 3558 and Identity 3571

### PR DESCRIPTION
**Pull request by cdwijayarathna**

### CODE REVIEW

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/resources/web/workflow-mgt/wf-request-list.jsp [hunk 1]
```diff
@@ -443,7 +444,8 @@
                            onclick="listWorkflows('<%=workflowReq.getRequestId()%>');return false;"
                            href="#" style="background-image: url(images/list.png);"
                            class="icon-link"><fmt:message key='workflows'/></a>
-                        <% if (workflowReq.getStatus().equals("PENDING") && workflowReq.getCreatedBy().equals(loggedUser)) { %>
+                        <% if (workflowReq.getCreatedBy() != null && workflowReq.getStatus().equals("PENDING") &&
```
Comment (omindu) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/resources/web/workflow-mgt/wf-request-list.jsp]: Better to make ```"PENDING".equals(...)```

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/java/org/wso2/carbon/identity/workflow/mgt/ui/WorkflowAdminServiceClient.java [hunk 2]
```diff
@@ -272,9 +272,9 @@ public WorkflowEventDTO getEvent(String id) throws RemoteException {
     }
 
     public WorkflowRequestDTO[] getRequestsCreatedByUser(String user, String beginDate, String endDate, String
-            dateCategory) throws RemoteException, WorkflowAdminServiceWorkflowException {
+            dateCategory, String status) throws RemoteException, WorkflowAdminServiceWorkflowException {
 
-        WorkflowRequestDTO[] requestDTOs = stub.getRequestsCreatedByUser(user, beginDate, endDate, dateCategory);
+        WorkflowRequestDTO[] requestDTOs = stub.getRequestsCreatedByUser(user, beginDate, endDate, dateCategory,status);
```
Comment (omindu) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/java/org/wso2/carbon/identity/workflow/mgt/ui/WorkflowAdminServiceClient.java]: check formatting.

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java [hunk 3]
```diff
@@ -525,4 +544,113 @@ public void updateStatusOfRequest(String requestId, String newState) throws Work
...
+                    while (iterator.hasNext()) {
+                        OMElementImpl child = iterator.next();
+                        checkMatchingTaskAndDelete(requestId, stub, arr, j, child);
+                    }
+
+                }
+            }
+        } catch (MalformedURLException | XMLStreamException | IllegalOperationFault | IllegalAccessFault |
+                RemoteException | IllegalStateFault | IllegalArgumentFault e) {
+            throw new WorkflowException("Error while deleting the human tasks of the request.");
```
Comment (omindu) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java]: Either log or wrap the error

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java [hunk 4]
```diff
@@ -525,4 +544,113 @@ public void updateStatusOfRequest(String requestId, String newState) throws Work
...
+            HumanTaskClientAPIAdminStub stub = null;
+            TSimpleQueryInput input = new TSimpleQueryInput();
+            TStatus reservedState = new TStatus();
+            reservedState.setTStatus(WorkFlowConstants.HT_STATE_RESERVED);
+            input.addStatus(reservedState);
+            TStatus readyState = new TStatus();
+            readyState.setTStatus(WorkFlowConstants.HT_STATE_READY);
+            input.addStatus(readyState);
+            input.setPageSize(100000);
+            input.setPageNumber(0);
```
Comment (omindu) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java]: Better to use constants.

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java (hunk 3 above)
Comment (cdwijayarathna) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java]: Already Fixed with https://github.com/wso2/carbon-identity/commit/a3d18711e29c26f313b8c8efd9c17302e3434eb1

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java (hunk 4 above)
Comment (cdwijayarathna) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java]: Thanks, Fixed with https://github.com/wso2/carbon-identity/pull/1675

**Pull request merged**